Add docs for nack backoff policy (#12660)
apache/pulsar: site2/docs/client-libraries-java.md

`autoUpdatePartitions`|boolean|If `autoUpdatePartitions` is enabled, a consumer subscribes to partition increasement automatically.<br /><br />**Note**: this is only for partitioned consumers.|true
`replicateSubscriptionState`|boolean|If `replicateSubscriptionState` is enabled, a subscription state is replicated to geo-replicated clusters.|false

> `negativeAckRedeliveryBackoff`|NegativeAckRedeliveryBackoff|Interface for custom message is negativeAcked policy, users can specify a `NegativeAckRedeliveryBackoff` for a consumer.| `NegativeAckRedeliveryExponentialBackoff`

Reviewer: what is the meaning of "interface xx is xx policy" (interface is policy?)?
```suggestion
`negativeAckRedeliveryBackoff`|NegativeAckRedeliveryBackoff|Interface for custom message is negativeAcked policy. You can specify `NegativeAckRedeliveryBackoff` for a consumer.| `NegativeAckRedeliveryExponentialBackoff`
```

>     .build();
> ```

> ### NegativeAckRedeliveryBackoff

Reviewer: Does this belong to`Negative Ack` in [this sheet](https://docs.google.com/spreadsheets/d/1YHYTkIXR8-Ql103u-IMI18TXLlGStK8uJjDsOOA0T20/edit#gid=1784579914&range=B73:C73)?
If so, how about changing the heading to `Negative acknowledgment`?

> The `NegativeAckRedeliveryBackoff` is to introduce a redelivery backoff mechanism which we can achieve redelivery with different delays according to the redeliveryCount the message is retried. 

Reviewer: ```suggestion
The `NegativeAckRedeliveryBackoff` introduces a redelivery backoff mechanism. You can achieve redelivery with different delays by setting `redeliveryCount ` of messages.
```
do you mean this?

```java
Consumer consumer =  client.newConsumer()
        .topic("my-topic")
        .subscriptionName("my-subscription")
        .negativeAckRedeliveryBackoff(NegativeAckRedeliveryExponentialBackoff.builder()
                .minNackTimeMs(1000)
                .maxNackTimeMs(60 * 1000)
                .build())
        .subscribe();
```

> > Notice: 

Reviewer: ```suggestion
> **Note**
```

> >   1. The negativeAckRedeliveryBackoff will not work with `consumer.negativeAcknowledge(MessageId messageId)` because we are not able to get the redelivery count from the message ID.

Reviewer: ```suggestion
>   - The `negativeAckRedeliveryBackoff` does not work with `consumer.negativeAcknowledge(MessageId messageId)` because you are not able to get the redelivery count from the message ID.
```

> >   2. The consumer crashes will trigger the redelivery of the unacked message, this case will not respect the `NegativeAckRedeliveryBackoff`, which means the message might get redelivered earlier than the delay time from the backoff.

Reviewer: ```suggestion
>   - If a consumer crashes, it triggers the redelivery of unacked messages. In this case, `NegativeAckRedeliveryBackoff` does not take effect and the messages might get redelivered earlier than the delay time from the backoff.
```

do you mean this?
